src/nerve_tools/configure_nerve.py

```diff
@@ -395,21 +434,23 @@ def update_subconfiguration_for_here(
         )
     for (service_name, service_info) in paasta_services:
         envoy_service_info: Optional[ServiceInfo] = None
-        if envoy_listeners:
-            key = f"{service_name}.{service_info['port']}"
-            if key in envoy_listeners:
-                envoy_service_info = copy.deepcopy(service_info)
-                envoy_ingress_port = envoy_listeners[key]
-                envoy_service_info['port'] = envoy_ingress_port
-                envoy_service_info['healthcheck_port'] = envoy_ingress_port
-                healthcheck_headers = envoy_service_info.get('extra_healthcheck_headers', {})
-                healthcheck_headers['Host'] = service_name
-                envoy_service_info['extra_healthcheck_headers'] = healthcheck_headers
+        envoy_key = f"{service_name}.{service_info['port']}"
+        if envoy_listeners and envoy_key in envoy_listeners:
+            service_info_copy = copy.deepcopy(service_info)
+            envoy_ingress_port = envoy_listeners[envoy_key]
+            healthcheck_headers = service_info_copy.get('extra_healthcheck_headers', {})
+            healthcheck_headers['Host'] = service_name
+            service_info_copy.update({
+                'port': envoy_ingress_port,
+                'healthcheck_port': envoy_ingress_port,
+                'extra_healthcheck_headers': healthcheck_headers,
+            })
+
         update_subconfiguration_for_here(
             service_name=service_name,
             service_info=cast(ServiceInfo, service_info),
             service_weight=10,
-            envoy_service_info=cast(ServiceInfo, envoy_service_info),
+            envoy_service_info=cast(Optional[ServiceInfo], envoy_service_info),
```

Comment:
Actually, I don't think you need a `cast` here because you declare the type on line 436. Unless I don't understand what `cast` does